metals/src/main/scala/scala/meta/internal/metals/WorkspaceSymbolQuery.scala

```diff
@@ -14,19 +14,25 @@ import scala.meta.internal.metals.WorkspaceSymbolQuery.AlternativeQuery
  */
 case class WorkspaceSymbolQuery(
     query: String,
-    alternatives: Array[AlternativeQuery]
+    alternatives: Array[AlternativeQuery],
+    isTrailingDot: Boolean
 ) {
   def matches(bloom: BloomFilter[CharSequence]): Boolean =
     alternatives.exists(_.matches(bloom))
   def matches(symbol: CharSequence): Boolean =
-    alternatives.exists(_.matches(symbol))
+    alternatives.exists(_.matches(symbol, isTrailingDot))
 }
 
 object WorkspaceSymbolQuery {
   def fromTextQuery(query: String): WorkspaceSymbolQuery = {
+    val isTrailingDot = query.endsWith(".")
+    val actualQuery =
+      if (isTrailingDot) query.stripSuffix(".")
+      else query
```

Comment:
isn't this equivalent to just

```scala
val actualQuery = query.stripSuffix(".")
```

?

Reply:
It is indeed, but I like it like this to stress the query changes only when there is a trailing `.`